api/service_broker.go

```diff
@@ -154,3 +143,27 @@ func serviceBrokerDelete(w http.ResponseWriter, r *http.Request, t auth.Token) e
 	}
 	return err
 }
+
+func decodeServiceBroker(request *http.Request) (*service.Broker, error) {
+	var broker service.Broker
+	dec := form.NewDecoder(nil)
+	dec.IgnoreCase(true)
+	dec.IgnoreUnknownKeys(true)
+	if err := request.ParseForm(); err != nil {
+		return nil, fmt.Errorf("unable to parse form: %v", err)
+	}
+	if err := dec.DecodeValues(&broker, request.Form); err != nil {
+		return nil, fmt.Errorf("unable to parse broker: %v", err)
+	}
+	cacheStr := request.FormValue("Config.CacheExpiration")
```

Comment:
Maybe we could use custom decoding for the service.Broker type?
https://github.com/ajg/form#custom-marshaling

Reply:
I tried, but the support to custom marshaling is limited. I'd have to implement `encoding.TextUnmarshaler` in the `CacheExpiration` field itself, and it also doesn't allow setting the field to nil (which I do when this field is omitted)